src/mlpack/methods/ann/layer/adaptive_max_pooling.hpp

```diff
@@ -51,6 +51,12 @@ class AdaptiveMaxPooling
    * @param outputShape A two-value tuple indicating width and height of the output.
    */
   AdaptiveMaxPooling(const std::tuple<size_t, size_t>& outputShape);
+    
+  //Copy constructor
+  AdaptiveMaxPooling(const AdaptiveMaxPooling& pooling);
+    
+  //Move constructor
+  AdaptiveMaxPooling(AdaptiveMaxPooling&& pooling);
```

Comment:
You can create an assignment operator for both copy and move constructors as you can see in other examples.